### PR TITLE
[PORT] Emissivity Blockers

### PR DESCRIPTION
**Pull request by francinum**
## About The Pull Request
Includes:
https://github.com/tgstation/tgstation/pull/49454
https://github.com/tgstation/tgstation/pull/57934

## Why It's Good For The Game
Emissivity is now clipped instead of showing over every single thing in the game.

## Changelog
:cl: actioninja, Kylerace
fix: Glowing overlays no longer render above everything, and mobs and items properly block them
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
what does this even do or any of this mean?

**Comment by francinum** [comment]
makes this 
image
look like this
image

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -41,6 +41,49 @@
...
+
+
+/atom/movable/Initialize(mapload)
+	. = ..()
+	switch(blocks_emissive)
+		if(EMISSIVE_BLOCK_GENERIC)
+			var/mutable_appearance/gen_emissive_blocker = mutable_appearance(icon, icon_state, EMISSIVE_BLOCKER_LAYER, EMISSIVE_BLOCKER_PLANE)
+			gen_emissive_blocker.dir = dir
+			gen_emissive_blocker.alpha = alpha
+			gen_emissive_blocker.appearance_flags |= appearance_flags
```
Comment (PowerfulBacon) [inline code/game/atoms_movable.dm:57]: Do things inherit appearance flags?
Also RESET_ALPHA should stop things being very quiet

Comment (francinum) [reply code/game/atoms_movable.dm:57]: I have no idea

**Comment by DatBoiTim** [comment]
Huh so this is why the arcade overlays were weird

**Review by utilbird — APPROVED** [review]
Nothing has catastrophically failed

**Pull request merged**